sycl/source/detail/graph_impl.hpp

```diff
@@ -582,6 +582,18 @@ class graph_impl {
   void makeEdge(std::shared_ptr<node_impl> Src,
                 std::shared_ptr<node_impl> Dest);
 
+  /// Throws an invalid exception if this function called
+  /// while the graph is recording a queue.
+  /// @param ExceptionMsg Message to append to the exception message
+  void throwIfGraphRecordingQueue(const std::string ExceptionMsg) {
```

Comment:
Can this be `const`?
```suggestion
  void throwIfGraphRecordingQueue(const std::string ExceptionMsg) const  {
```